### PR TITLE
Reduced the number of docker images by one `pybamm:latest`

### LINKED ISSUE #3666: Reduce the number of Docker Images to just one image
Currently we build and push 5 variants of Docker Images regularly i.e. `latest`, `jax`, `odes`, `idaklu` & `all`.
The idea behind this issue is to have a single standard image variant to be updated regularly to save resources and make things simple for developers.
As discussed in #3661

### PR DESCRIPTION
**Pull request by santacodes**
# Description

The `Dockerfile` is reduced to include all the solvers by default, the documentation for `install from source (docker)` and `docker.yml` workflow file have been updated to push only one single docker image tagged `pybamm:latest` to dockerhub with all the solvers contained in them by default, i.e `IDAKLU` and `JAX`.

Fixes #3666 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: docs/source/user_guide/installation/install-from-docker.rst
Comment (agriyakhetarpal) [inline docs/source/user_guide/installation/install-from-docker.rst:184]: Could we keep these two instructions?

File: docs/source/user_guide/installation/install-from-docker.rst
Comment (agriyakhetarpal) [inline docs/source/user_guide/installation/install-from-docker.rst:101]:
```suggestion
  PyBaMM's Docker image comes with all available solvers by default. These solvers include ``IDAKLU`` IDAS solver provided by the SUNDIALS linked with SuiteSparse's KLU and the ``JAX`` solver.
```

A small comment on the wording here. The IDAKLU solver is the IDA solver with sensitivities, hence the extra S ([`IDAS`](https://sundials.readthedocs.io/en/latest/idas/)). SUNDIALS is not shipped with KLU by default and is provided by SuiteSparse.

**Comment by agriyakhetarpal** [comment]
Also, we'll need a CHANGELOG entry about this breaking change, mentioning the removal of the other arguments and asking users to use the `pybamm/pybamm:latest` image instead.

File: docs/source/user_guide/installation/install-from-docker.rst
Comment (santacodes) [reply docs/source/user_guide/installation/install-from-docker.rst:184]: I think we already have those instructions in the above documentation block (`building docker image locally from source`)

**Comment by santacodes** [comment]
Could you do a one final review? I will add a CHANGELOG after it.

File: docs/source/user_guide/installation/install-from-docker.rst
Comment (agriyakhetarpal) [reply docs/source/user_guide/installation/install-from-docker.rst:184]: Oops, missed those. Thanks!

File: CHANGELOG.md
Comment (agriyakhetarpal) [inline CHANGELOG.md:33]: Just rephrasing this sentence a bit to make it clearer + the PR number was incorrect

```suggestion
- Removed multiple Docker images. Here on, a single Docker image tagged `pybamm/pybamm:latest` will be provided with both solvers (`IDAKLU` and `JAX`) pre-installed. ([#3992](https://github.com/pybamm-team/PyBaMM/pull/3992))
```

**Review by agriyakhetarpal — APPROVED** [review]
Thanks, @santacodes! This looks good to me. I'll leave it up to @arjxn-py to take another look, after which we can merge.

File: .github/workflows/docker.yml
Comment (arjxn-py) [inline .github/workflows/docker.yml:18]: I'd suggest to only remove - 
```
matrix:
    build-args: ["No solvers", "JAX", "IDAKLU", "ALL"]
```

File: scripts/Dockerfile
Comment (arjxn-py) [inline scripts/Dockerfile:37]: I hope you've created docker image using this updated file & have tested for available solvers inside the running containers with commands i.e. `pybamm.have_idaklu()` & `pybamm.have_jax()`

File: docs/source/user_guide/installation/install-from-docker.rst
Comment (arjxn-py) [inline docs/source/user_guide/installation/install-from-docker.rst:39]: As `:latest` is our only and default tag, user won't need to specify it explicitly for now, hence we can keep it like this - 
```suggestion
    docker run -it pybamm/pybamm
```

File: docs/source/user_guide/installation/install-from-docker.rst
Comment (arjxn-py) [inline docs/source/user_guide/installation/install-from-docker.rst:26]: Ditto
```suggestion
    docker pull pybamm/pybamm
```

**Review by arjxn-py — COMMENTED** [review]
Looks good @santacodes, just a couple of comments & suggestions below.

File: .github/workflows/docker.yml
Comment (agriyakhetarpal) [reply .github/workflows/docker.yml:18]: We'll have to remove the `strategy:` as well because that is applicable just for job matrices

File: scripts/Dockerfile
Comment (santacodes) [reply scripts/Dockerfile:37]: Yes I did and both of the solvers are available I tested it out on docker interactive session

**Comment by arjxn-py** [comment]
This one is good to merge too, I'll merge this as soon as tests pass

**Comment by santacodes** [comment]
Sounds good to me

**Review by arjxn-py — APPROVED** [review]
Thanks for taking this up @santacodes 🙂

**Pull request merged**
